crates/rpc/rpc/src/eth/helpers/signer.rs

```diff
@@ -41,23 +41,24 @@ impl DevSigner {
     pub fn random_signers(num: u32) -> Vec<Box<dyn EthSigner + 'static>> {
         let mut signers = Vec::new();
         for _ in 0..num {
-            let (sk, pk) = secp256k1::generate_keypair(&mut rand::thread_rng());
+            let sk = PrivateKeySigner::random_with(&mut rand::thread_rng());
 
-            let address = reth_primitives::public_key_to_address(pk);
+            let address = sk.address();
             let addresses = vec![address];
+
             let accounts = HashMap::from([(address, sk)]);
             signers.push(Box::new(Self { addresses, accounts }) as Box<dyn EthSigner>);
         }
         signers
     }
 
-    fn get_key(&self, account: Address) -> Result<&SecretKey> {
+    fn get_key(&self, account: Address) -> Result<&PrivateKeySigner> {
         self.accounts.get(&account).ok_or(SignError::NoAccount)
     }
 
     fn sign_hash(&self, hash: B256, account: Address) -> Result<Signature> {
-        let secret = self.get_key(account)?;
-        let signature = sign_message(B256::from_slice(secret.as_ref()), hash);
+        let sk = self.get_key(account)?.to_field_bytes();
+        let signature = sign_message(B256::from_slice(sk.as_ref()), hash);
         signature.map_err(|_| SignError::CouldNotSign)
```

Comment:
you can use the signer API here:

use alloy_signer::signer::SignerSync;

let sk = self.get_key(account)?.